File: JAVA/1394. Find Lucky Integer in an Array.java
class Solution {
    public int findLucky(int[] arr) {
        Map<Integer,Integer>m=new HashMap<>();
        for(int i:arr){
            m.put(i, m.getOrDefault(i, 0) + 1);
        }
        int v=-1;
        for (Map.Entry<Integer,Integer> e : m.entrySet()){
            if(e.getValue()==e.getKey()){
                if(v<e.getValue())
                v=e.getKey();
            }
        }
        return v;
    }
}

File: C++/1394. Find Lucky Integer in an Array.cpp
class Solution {
public:
    int findLucky(vector<int>& a) {
        map<int,int>m;
        int v=-1;
        for(int i:a){
            m[i]++;
        }
        for(const auto& p:m){
            if(p.first==p.second){
                if(p.first>v)
                v=p.first;
            }
        }
        return v;
    }
};